subsys/bluetooth/host/hci_core.c

```diff
@@ -2176,7 +2176,7 @@ static void hci_encrypt_change(struct net_buf *buf)
 			 * Start SMP over BR/EDR if we are pairing and are
 			 * central on the link
 			 */
-			if (atomic_test_bit(conn->flags, BT_CONN_BR_PAIRING) &&
+			if (atomic_test_bit(conn->flags, BT_CONN_BR_PAIRED) &&
```

Comment:
Is this really right? Shouldn't you also check if we're already paired on the LE side? If so, then we should probably not start SMP. More generally, we have existing APIs for managing pairing-related keys, so wouldn't it make sense to check for the existence of those keys rather than have a separate (and seemingly redundant) `PAIRED` flag?

Reply:
The code block is used to derive LE LTK on BR ACL connection.
`BT_CONN_BR_PAIRED` is used to flag the BR pairing is done in current connection. And then cross transport procedure can be started.

Reply:
Is it possible that there's a bug with clearing the `PAIRING` flag? Maybe it should only be cleared when we get a successful encrypt change event? In that case you might not need a new flag.

Reply:
No. After pairing is completed, encryption is not necessary.

There is a test case that the pairing is initialised by PTS. After the pairing is done, the encryption needs to be started by IUT.